parachain/pallets/commitments/src/lib.rs

```diff
@@ -98,69 +106,145 @@ decl_module! {
 }
 
 impl<T: Config> Module<T> {
+	// Generate a key for basic offchain storage
+	fn offchain_key_basic(account_id: T::AccountId, hash: H256) -> Vec<u8> {
+		(T::INDEXING_PREFIX, ChannelId::Basic, account_id, hash).encode()
+	}
 
-	// Generate a key for offchain storage
-	fn offchain_key(channel_id: ChannelId, hash: H256) -> Vec<u8> {
-		(T::INDEXING_PREFIX, channel_id, hash).encode()
+	// Generate a key for incentivized offchain storage
+	fn offchain_key_incentivized(hash: H256) -> Vec<u8> {
+		(T::INDEXING_PREFIX, ChannelId::Incentivized, hash).encode()
 	}
 
 	// TODO: return proper weight
 	fn commit() -> Weight {
-		let mut weight: Weight = 0;
+		Self::commit_for_basic_channel() + Self::commit_for_incentivized_channel()
+	}
 
-		for channel_id in ChannelId::into_enum_iter() {
-			weight += Self::commit_for_channel(channel_id);
+	fn commit_for_basic_channel() -> Weight {
```

Comment:
I think we're at the point where this commitments pallet has outlived its usefulness, as it was originally designed to generate one kind of commitment. Now its generating a basic and an incentivized commitment, which breaks abstraction boundaries.

Lets move the commitment code to the respective outbound channels, and then delete this pallet.

Reply:
Good point, I like the idea.